src/main/java/gui/TimetableBlock.java

```diff
@@ -0,0 +1,198 @@
+package gui;
+
+import javafx.event.EventHandler;
+import javafx.geometry.Point2D;
+import javafx.scene.Parent;
+import javafx.scene.input.MouseEvent;
+import javafx.scene.layout.Region;
+
+/**
+ * Class that resembles a draggable, resiable block inside the timetable.
+ * Highly volatile. Do not poke too much.
+ */
+class TimetableBlock extends Region {
+
+    private String normalStyle = "-fx-border-style: solid inside;"
+            + "-fx-border-width: 3;"
+            + "-fx-border-color: yellow;"
+            + "-fx-background-color: orange;";
+    private String dragStyle = "-fx-border-style: solid inside;"
+            + "-fx-border-width: 3;"
+            + "-fx-border-color: red;"
+            + "-fx-background-color: orange;";
+
+    private double margin = 6.0; // margin of edge size for dragging.
+    private int dragging = 0; // 0 is center, 1 top, 2 right, 3 bottom, 4 left.
+
+    private double mouseCurrentXPosition;
+    private double mouseCurrentYPosition;
+    private double mouseCurrentXMovement;
+    private double mouseCurrentYMovement;
+
+    private Point2D scrollMouse;
+
+    /**
+     * Constructor class for timetable block.
+     * @param pane parent area this block is located in.
+     */
+    TimetableBlock(RootCenterArea pane) {
+        // some basic temporary styling for the block. That's size, position, and style.
+        setLayoutX(100.0);
+        setLayoutY(100.0);
+        setMinHeight(100.0);
+        setMinWidth(100.0);
+        setStyle(normalStyle);
+
+        // mouse press handler. Called when mouse is pressed on this block.
+        setOnMousePressed(new EventHandler<MouseEvent>() {
+            @Override
+            public void handle(MouseEvent event) {
+                setStyle(dragStyle);
+                dragging = findEdgeZone(event);
+                mouseCurrentXPosition = event.getSceneX();
+                mouseCurrentYPosition = event.getSceneY();
+            }
+        });
+
+        // mouse drag handler (uses helper function because it's too large)
+        setOnMouseDragged(new EventHandler<MouseEvent>() {
+            @Override
+            public void handle(MouseEvent event) {
+                onMouseDraggedHelper(pane, event);
+            }
+        });
+
+        // mouse release handler: called when mouse is released on this block.
+        setOnMouseReleased(new EventHandler<MouseEvent>() {
+            @Override
+            public void handle(MouseEvent event) {
+                setStyle(normalStyle);
+            }
+        });
+    }
+
+    /**
+     * Helper function for MouseDragged event.
+     * @param pane pane this block is located in.
+     * @param event the mousedrag event in question.
+     */
+    private void onMouseDraggedHelper(RootCenterArea pane, MouseEvent event) {
```

Comment:
In the constructor you now make new eventhandler and call this method in the body, why not return an actual eventhandler from this function and add it like: "setOnMouseReleased(onMouseDraggedHelper())"m seems a lot cleaner to me